Tests/ProductsControllerTest.cs

```diff
@@ -31,9 +33,127 @@ public void Index_Should_Return_As_Expected(List<ProductDetailDto> expected)
 
         var actual = model;
 
-        for (int i = 0; i < actual.Count; i++)
+        for (int i = 0; i < expected.Count; i++)
         {
-            Assert.Equal(expected[i], actual[i]);
+            Assert.Equal(expected[i].Name, actual[i].Name);
+            Assert.Equal(expected[i].SupplierName, actual[i].SupplierName);
+            Assert.Equal(expected[i].CategoryName, actual[i].CategoryName);
+            Assert.Equal(expected[i].ImageUrl, actual[i].ImageUrl);
+            Assert.Equal(expected[i].Price, actual[i].Price);
         }
     }
+
+    [Theory, AutoMoqData]
+    public void Index_ShouldThrowExceptionAndReturnErrorDetails_WhenErrorOccurs(AggregateException expected) //ErrorDetails
+    {
+        // Arrange
+        var configurationMock = new Mock<IConfiguration>();
+        var productServiceMock = new Mock<IProductService>();
+
+        var sut = new ProductsController(productServiceMock.Object, configurationMock.Object);
+        productServiceMock.Setup(productService => productService.GetProductDetailsAsync().Exception).Returns(expected);
```

Comment:
Not a correct setup for throwing an error, I shared with you a link how to perform it correctly